Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -162,11 +162,18 @@ open class NavigationMapView: UIView {
     var routePoints: RoutePoints?
     var routeLineGranularDistances: RouteLineGranularDistances?
     var routeRemainingDistancesIndex: Int?
-    var routeLineTracksTraversal: Bool = false
     var fractionTraveled: Double = 0.0
     var currentLegIndex: Int?
     var currentLegCongestionLevels: [CongestionLevel]?
     var currentLineGradientStops = [Double: UIColor]()
+    var routeLineTracksTraversal: Bool = false {
+        didSet {
+            if routeLineTracksTraversal, let route = self.routes?.first {
```

Comment:
- does it mean that if routes is set after `routeLineTracksTraversal` this code won't work? 
- what if `routeLineTracksTraversal` changed when navigation started, will it work correctly?

Reply:
1. Yes, if routes set after `routeLineTracksTraversal `, it won't be triggered. To store the route coordinates points for vanishing route line, by logic it would call the `navigationMapView.show(_:legIndex:)` show the route line first, and then it would check the `routeLineTracksTraversal` and decide whether to store the route points.
2. Good question. If during navigation session and this parameter changes to `true`, it works. I'll add the function to disable this feature when it sets to `false` during navigation session.

Reply:
The gif above tested the parameter value change during navigation.

Reply:
Thanks for the gif, looks good to me!